### ISSUE TITLE
Wrong libusb v1.0.25 in v3.1.0

### ISSUE THREAD

**Issue by pgorgon-hem** [vpelletier/python-libusb1#101]
Hello,

In release notes it states that latest release v3.1.0 contains libusb in version 1.0.26, which is incorrect, it has 1.0.25. Any chance for update?

**Comment by vpelletier** [comment]
Thank you for your report. This is supposed to not happen thanks to sha256 check of libusb 7z archive, both to detect new releases when packaging and to check what was actually included...

Except:
- this is from 12b6639d48b7cc2f2fdb216a8ba7644aa5e1cd96, which updates the sha256:
```diff
-    '3d1c98416f454026034b2b5d67f8a294053898cb70a8b489874e75b136c6674d'
+    '9c242696342dbde9cdc47239391f71833939bf9f7aa2bbb28cdaabe890465ec5'
```
- these are the neighbouring 7z currently available for download:
```
$ sha256sum libusb-1.0.2?.7z
3d1c98416f454026034b2b5d67f8a294053898cb70a8b489874e75b136c6674d  libusb-1.0.25.7z
8ce6b97d2fe526bd9539166c2cfe01a1d28bcf972d9bdd7c37911df67eeec86f  libusb-1.0.26.7z
19835e290f46fab6bd8ce4be6ab7dc5209f1c04bad177065df485e51dc4118c8  libusb-1.0.27.7z
```

Old sha256 matches 1.0.25, but new one does not match 1.0.26. Which means libusb1 modified that 7z after initially releasing it - maybe they bundled the wrong version ?

So, indeed, a new release on my side is necessary to fix that.

**Comment by vpelletier** [comment]
BTW, if you have suggestions on ways to automate (on GNU/Linux) the detection of the actual libusb1 dlls, I would be interested to include this to the release process, to avoid recurrence of this issue.

**Comment by vpelletier** [comment]
Should be fixed by the latest release.

**Comment by pgorgon-hem** [comment]
Thanks a lot for a fix!
Do I understand correctly that you don't know how to check actual version of dll? I have no experience with it, but I suppose we could compile simple windows executable which loads libusb dll and uses standard libusb api call to get version and print it. Then it could be run in wine. But I'm not sure if it can run e.g. in docker.

**Comment by vpelletier** [comment]
> Do I understand correctly that you don't know how to check actual version of dll?

Correct, within the limitation that I am packaging on GNU/Linux and hence the packaging-time check should be able to work on it.

For a low-tech example, I checked whether the libusb version is available as an ascii string in the library - something I can trivially check. Unfortunately, no luck: there is no clear string containing this  info.

For another example, I tried dumping info from the dll using objdump, but I do not see the version info. I was under the impression the file information for a dll on Windows displayed some info about the dll, including a version... Am I misremembering, or is it just not present for libusb1 ?
